src/background.ts

```diff
@@ -73,20 +74,20 @@ async function generateQuery(message: { context: string, url: string, dims: [num
         } else {
           throw resp;
         }
-    }).then(async(blobFile) => {
-      graphicWidth = message.dims[0];
-      graphicHeight = message.dims[1];
-      if(graphicWidth> 1200 && graphicWidth > graphicHeight){
-        message.dims[0] = 1200;
-        message.dims[1] = Math.round(graphicHeight*1200/graphicWidth);
-        return fromBlob(blobFile, message.dims[0], 'auto', 'webp');
-      } else if(graphicHeight > 1200){
-        message.dims[0] = Math.round(graphicWidth*1200/graphicHeight);
-        message.dims[1] = 1200;
-        return fromBlob(blobFile, message.dims[0], 'auto', 'webp');
-      } else {
-        return blobFile;
+    }).then(async(blob: Blob) => {
+      const blobFile = new File([blob], "buffer.jpg", {type: blob.type});
+      console.log('originalFile instanceof Blob', blobFile instanceof Blob); // true
+      console.log(`originalFile size ${blobFile.size / 1024 / 1024} MB`);
+      const options = {
+        maxSizeMB: 5,
```

Comment:
Do we know how this library prioritizes lowering resolution vs. doing other compression? I think for our purposes, we would want it to first lower the resolution, then only adjust the (lossy) compression if necessary to then stay under the overall MB limit.

Reply:
From the library, I think the default compression will "only reduce quality, but always keep width and height". https://github.com/Donaldcwl/browser-image-compression